Jenkinsfile

```diff
@@ -18,315 +18,326 @@ pipeline {
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-master-arc.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						    sh("kubectl -n dev cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-master-arc.xml pytest-master-arc.xml")
+						    junit "pytest-master-arc.xml"
 						}
-						junit 'pytest-master-arc.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Zero-File-Creation') {
+		stage("Test-Zero-File-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-zero-file.xml ' +
-								'-m zero_file /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+                    withKubeConfig([credentialsId: "prod-kube-config"]) {
+                        sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+                            "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+                            "--junitxml=/nres/code/pytest-zero-file.xml -m zero_file /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-zero-file.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						    sh("kubectl -n dev cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-zero-file.xml pytest-zero-file.xml")
+						    junit "pytest-zero-file.xml"
 						}
-						junit 'pytest-zero-file.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Science-File-Creation') {
+		stage("Test-Science-File-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-science-files.xml ' +
-								'-m science_files /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+			        withKubeConfig([credentialsId: "prod-kube-config"]) {
+						sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+                            "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+                            "--junitxml=/nres/code/pytest-science-files.xml -m science_files /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-science-files.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
```

Comment:
If all these steps use the `dev` namespace, you can use the `dev-kube-config` credentials.

Reply:
Fixed